core/imds/src/main/java/software/amazon/awssdk/imds/internal/DefaultEc2Metadata.java

```diff
@@ -86,191 +88,117 @@ public Ec2Metadata.Builder toBuilder() {
                         .endpoint(endpoint)
                         .tokenTtl(tokenTtl)
                         .endpointMode(endpointMode)
-                        .httpDebugOutput(httpDebugOutput)
                         .httpClient(httpClient);
     }
 
-    @Override
-    public boolean equals(Object obj) {
-        if (this == obj) {
-            return true;
-        }
-        if (obj == null || getClass() != obj.getClass()) {
-            return false;
-        }
-        DefaultEc2Metadata ec2Metadata = (DefaultEc2Metadata) obj;
-
-        if (!Objects.equals(retryPolicy, ec2Metadata.retryPolicy)) {
-            return false;
-        }
-        if (!Objects.equals(endpoint, ec2Metadata.endpoint)) {
-            return false;
-        }
-        if (!Objects.equals(tokenTtl, ec2Metadata.tokenTtl)) {
-            return false;
-        }
-        if (!Objects.equals(endpointMode, ec2Metadata.endpointMode)) {
-            return false;
-        }
-        if (!Objects.equals(httpDebugOutput, ec2Metadata.httpDebugOutput)) {
-            return false;
-        }
-        return Objects.equals(httpClient, ec2Metadata.httpClient);
-    }
-
-    @Override
-    public int hashCode() {
-
-        int result = retryPolicy != null ? retryPolicy.hashCode() : 0;
-        result = 31 * result + (endpoint != null ? endpoint.hashCode() : 0);
-        result = 31 * result + (tokenTtl != null ? tokenTtl.hashCode() : 0);
-        result = 31 * result + (endpointMode != null ? endpointMode.hashCode() : 0);
-        result = 31 * result + (httpDebugOutput != null ? httpDebugOutput.hashCode() : 0);
-        result = 31 * result + (httpClient != null ? httpClient.hashCode() : 0);
-
-        return result;
-    }
-
-    @Override
-    public String toString() {
-        return "DefaultEc2Metadata{" +
-               "retries=" + retryPolicy +
-               ", endpoint='" + endpoint + '\'' +
-               ", tokenTtl=" + tokenTtl +
-               ", endpointMode='" + endpointMode + '\'' +
-               ", httpDebugOutput='" + httpDebugOutput + '\'' +
-               ", httpClient= " + httpClient.clientName() +
-               '}';
-    }
-
     /**
-     * Gets the specified instance metadata value by the given path.
-     * @param path  Input path
+     * Gets the specified instance metadata value by the given path. Will retry base on the {@link Ec2MetadataRetryPolicy retry
+     * policy} provided, in the case of an IOException during request. Will not retry on SdkClientException, like 4XX/5XX HTTP
```

Comment:
Should we retry 5XX? sorry, didn't notice this earlier

Reply:
no you are right, we should